File: 백준/Silver/4948. 베르트랑 공준/베르트랑 공준.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        StringBuilder sb = new StringBuilder();
        while (true) {
            int count = 0;
            int n = Integer.parseInt(br.readLine());
            if (n == 0) break;

            for (int i = n + 1; i <= 2 * n; i++) {
                if (isPrime(i)) {
                    count++;
                }
            }
            sb.append(count).append("\n");
        }

        bw.write(sb.toString());
        bw.flush();
        bw.close();
    }

    private static boolean isPrime(int num) {
        if (num < 2) {
            return false;
        } else if (num == 2) {
            return true;
        } else if (num % 2 == 0) {
            return false;
        } else {
            for (long i = 3; i <= Math.sqrt(num); i += 2) {
                if(num % i == 0) {
                    return false;
                }
            }
            return true;
        }
    }
}

File: 백준/Silver/4948. 베르트랑 공준/베르트랑 공준.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import kotlin.math.sqrt

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))

    val sb = StringBuilder()
    while (true) {
        var count = 0
        val n = br.readLine().toInt()
        if (n == 0) break

        for (i in n + 1 .. 2 * n) {
            if (isPrime(i)) {
                count++
            }
        }
        sb.append(count).append("\n")
    }

    bw.write(sb.toString())
    bw.flush()
    bw.close()
}

fun isPrime(num: Int): Boolean {
    if (num < 2) {
        return false
    }else if (num == 2) {
        return true
    }else if (num % 2 == 0) {
        return false
    } else {
        for(i in 3..sqrt(num.toDouble()).toInt() step 2) {
            if (num % i == 0) {
                return false
            }
        }
        return true
    }
}